### PR TITLE
Refactor update codemeta

### PR DESCRIPTION
**Pull request by kjgarza**
Do not merge until after F2F meeting.

fix https://github.com/datacite/bolognese/issues/30

### CODE REVIEW

**Comment by mfenner** [comment]
@kjgarza is the main change agents -> authors and title -> name, aligning with schema.org in both cases? Anything else that has changed?

File: lib/bolognese/readers/codemeta_reader.rb [hunk 1]
```diff
@@ -36,7 +36,9 @@ def read_codemeta(string: nil, **options)
         id = Array.wrap(identifiers).first.to_h.fetch("identifier", nil)
         doi = Array.wrap(identifiers).find { |r| r["identifierType"] == "DOI" }.to_h.fetch("identifier", nil)
 
-        creators = get_authors(from_schema_org_creators(Array.wrap(meta.fetch("agents", nil))))
+        creators =  meta.fetch("agents", nil).nil? ? get_authors(from_schema_org_creators(Array.wrap(meta.fetch("authors", nil)))) : get_authors(from_schema_org_creators(Array.wrap(meta.fetch("agents", nil))))
```
Comment (mfenner) [inline lib/bolognese/readers/codemeta_reader.rb]: This doesn't look right. What are you trying to do here?

File: lib/bolognese/readers/codemeta_reader.rb [hunk 2]
```diff
@@ -58,12 +60,14 @@ def read_codemeta(string: nil, **options)
           { "subject" => s }
         end
 
+        titles =  meta.fetch("title", nil).nil? ?  [{ "title" => meta.fetch("name", nil) }] : [{ "title" => meta.fetch("title", nil) }]
```
Comment (mfenner) [inline lib/bolognese/readers/codemeta_reader.rb]: Again, I don't understand this use of the terniary operator.

Comment (kjgarza) [reply lib/bolognese/readers/codemeta_reader.rb]: I think `title` doesn't exist in codemeta-v2 (https://raw.githubusercontent.com/codemeta/codemeta/2.0/codemeta.jsonld) but it does in codemeta-v1  (https://raw.githubusercontent.com/codemeta/codemeta/1.0/codemeta.jsonld). I wanted to account for reading metadata in both versions.

**Pull request merged**